src/lib/util/src/mmio.rs

```diff
@@ -0,0 +1,40 @@
+use core::ptr::{read_volatile, write_volatile};
+
+pub fn read8(address: usize) -> u8 {
+    unsafe { read_volatile(address as *mut u8) }
+}
+
+pub fn write8(address: usize, value: u8) {
+    unsafe { write_volatile(address as *mut u8, value) }
+}
+
+pub fn read16(address: usize) -> u16 {
+    unsafe { read_volatile(address as *mut u16) }
+}
+
+pub fn write16(address: usize, value: u16) {
+    unsafe { write_volatile(address as *mut u16, value) }
+}
+
+pub fn read32(address: usize) -> u32 {
+    unsafe { read_volatile(address as *mut u32) }
+}
+
+pub fn write32(address: usize, value: u32) {
+    unsafe { write_volatile(address as *mut u32, value) }
+}
+
+// NOTE: Some hardware does not like direct u64 MMIO access.
+// So read low and high word (little endian) one at a time, then merge them.
+pub fn read64(reg: usize) -> u64 {
```

Comment:
This is going to burn you at some point, the problem with this is that some hardware will discard high order bits when you read32 the low 32 bits ... it can really suck that way. I wonder if you want to make this explicit, like
read64_hl
or something. 
i.e. make it really clear in the name that this does not do a real read64?

Reply:
Alright, I'll do that.

Reply:
I will go with `read64le`.